mpcite/models.py

```diff
@@ -0,0 +1,378 @@
+        return material.formula_pretty
+
+
+class ExplorerGetJSONResponseModel(BaseModel):
+    osti_id: str
+    title: str
+    report_number: str
+    doi: str
+    product_type: str
+    language: str
+    country_publication: str
+    description: str
+    site_ownership_code: str
+    publication_date: str
+    entry_date: str
+    contributing_organizations: str
+    authors: List[str]
+    subjects: List[str]
+    contributing_org: str
+    doe_contract_number: str
+    sponsor_orgs: List[str]
+    research_orgs: List[str]
+    links: List[Dict[str, str]]
+
+
+class LogContent(BaseModel):
+    date: datetime = Field(default=datetime.now())
+    last_updated_count: int = Field(default=0,
+                                    title="Number of DOIs that are updated",
+                                    description="Number of DOIs that are updated in this run. \nDOI is update iff"
+                                                "\n1) this is a new submission"
+                                                "\n2) bibtex has changed (from Explorer)"
+                                                "\n3) DOI number has changed(this should never happen, but for the "
+                                                "sake of fail safe)")
+    created_at_count: int = Field(default=0,
+                                  title="Number of records created",
+                                  description="Number of records created")
+    elsevier_updated_on_count: int = Field(default=0,
+                                           title="Number of DOIs uploaded to Elsevier",
+                                           description="Number of DOIs uploaded to Elsevier")
+    last_validated_count: int = Field(default=0,
+                                      title="number of DOIs validated",
+                                      description="Number of DOIs that are validated in this run")
+    material_data_base_count: int = Field(default=0,
+                                          title="Total number of materials in the database",
+                                          description="Total number of materials in the database")
+    doi_store_count: int = Field(default=0,
+                                 title="Number of DOIs that are in the DOI collection",
+                                 description="Number of DOIs that are in the DOI collection")
+    bibtex_count: int = Field(default=0,
+                              title="Number of DOIs that has bibtex",
+                              description="Number of DOIs that has bibtex ")
+    doi_completed: int = Field(default=0,
+                               title="Number of DOIs Completed",
+                               description="Number of DOIs that has a COMPLETED status")
+    doi_pending: int = Field(default=0,
+                             title="Number of DOI Pending",
+                             description="Number of DOIs that has PENDING status")
```

Comment:
What is this for?

Reply:
oops, sry, i had a log file, and it is used to define a model for the log

I tried to define a model for all the object, since I can simply use `myModel` with a `.` to access its field, it was very convenient way to debug and write things.

Reply:
it is now removed